File: C# Version/MenuProject/MenuItemForChildForm.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;

namespace MenuProject
{
    class MenuItemForChildForm : ToolStripMenuItem
    {
        new mainForm Parent;
        Type _childForm;

        public MenuItemForChildForm(mainForm inputMainForm, Type inputChildForm)
        {
            Parent = inputMainForm;
            _childForm = inputChildForm;

            //set the menuItem properties
            this.Name = inputChildForm.Name;
            this.Text = inputChildForm.Name;
            this.Click += childFormToolStripMenuItem_Click;
        }

        public Type childForm
        {
            get { return _childForm; }
        }

        private void childFormToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ToolStripMenuItem menuItem = sender as ToolStripMenuItem;

            dynamic child = Activator.CreateInstance(this._childForm);
            child.MdiParent = Parent;
            child.Show();
        }
    }
}

File: VB Version/VB MenuProject/MenuItemForChildForm.vb
Namespace MenuProject
     
    Class MenuItemForChildForm
        Inherits ToolStripMenuItem

        Shadows Parent As mainForm
        Private _childForm As Type

        Public Sub New(ByVal inputMainForm As mainForm, ByVal inputChildForm As Type)
            Parent = inputMainForm
            _childForm = inputChildForm

            'set the menuItem properties
            Me.Name = inputChildForm.Name
            Me.Text = inputChildForm.Name
            AddHandler Me.Click, AddressOf childFormToolStripMenuItem_Click
        End Sub

        Public ReadOnly Property childForm() As Type
            Get
                Return _childForm
            End Get
        End Property

        Private Sub childFormToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim menuItem As ToolStripMenuItem = TryCast(sender, ToolStripMenuItem)

            Dim child = Activator.CreateInstance(Me._childForm)
            child.MdiParent = Parent
            child.Show()
        End Sub
    End Class

End Namespace


